Spigot-Server-Patches/0695-Fix-NotePlayEvent.patch

```diff
@@ -0,0 +1,124 @@
+index 454566fe8222684e39f8c40ad3e2714c4ed32699..1053ffe35c2a1cd0104c680f299ff2280ab12403 100644
+--- a/src/main/java/net/minecraft/world/level/block/state/properties/BlockPropertyInstrument.java
++++ b/src/main/java/net/minecraft/world/level/block/state/properties/BlockPropertyInstrument.java
+@@ -26,6 +26,7 @@ public enum BlockPropertyInstrument implements INamable {
+         return this.q;
+     }
+ 
++    public final SoundEffect getSoundEffect() { return this.b(); } // Paper - OBFHELPER
+     public SoundEffect b() {
+         return this.r;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+index 712a32c0b64dace1f3e3d6f8acb8cbef1736b1f9..183caeeff8b24e4ee6ee4f9025fb5ce56861038d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
++++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+@@ -12,6 +12,7 @@ import java.io.File;
+ import java.io.IOException;
+ import java.util.Arrays;
+ import java.util.Collections;
++import java.util.EnumMap;
+ import java.util.HashMap;
+ import java.util.List;
+ import java.util.Locale;
+@@ -35,10 +36,12 @@ import net.minecraft.util.datafix.fixes.DataConverterTypes;
+ import net.minecraft.world.item.Item;
+ import net.minecraft.world.level.block.Block;
+ import net.minecraft.world.level.block.state.IBlockData;
++import net.minecraft.world.level.block.state.properties.BlockPropertyInstrument;
+ import net.minecraft.world.level.material.FluidType;
+ import net.minecraft.world.level.storage.SavedFile;
+ import org.bukkit.Bukkit;
+ import org.bukkit.Fluid;
++import org.bukkit.Instrument;
+ import org.bukkit.Material;
+ import org.bukkit.NamespacedKey;
+ import org.bukkit.Registry;
+@@ -101,6 +104,7 @@ public final class CraftMagicNumbers implements UnsafeValues {
+     // Paper start
+     private static final Map<org.bukkit.entity.EntityType, net.minecraft.world.entity.EntityTypes<?>> ENTITY_TYPE_ENTITY_TYPES = new HashMap<>();
+     private static final Map<net.minecraft.world.entity.EntityTypes<?>, org.bukkit.entity.EntityType> ENTITY_TYPES_ENTITY_TYPE = new HashMap<>();
++    private static final Map<Instrument, BlockPropertyInstrument> INSTRUMENT_INSTRUMENT = new EnumMap<>(Instrument.class);
+ 
+     static {
+         for (org.bukkit.entity.EntityType type : org.bukkit.entity.EntityType.values()) {
+@@ -108,6 +112,9 @@ public final class CraftMagicNumbers implements UnsafeValues {
+             ENTITY_TYPE_ENTITY_TYPES.put(type, IRegistry.ENTITY_TYPE.get(CraftNamespacedKey.toMinecraft(type.getKey())));
+             ENTITY_TYPES_ENTITY_TYPE.put(IRegistry.ENTITY_TYPE.get(CraftNamespacedKey.toMinecraft(type.getKey())), type);
+         }
++        for (Instrument instrument : Instrument.values()) {
```

Comment:
anything against matching on name here, relying on ordinal values just creates issues and makes it harder to detect if stuff doesn't line up as intended, it's a static init so not really a pressing issue if it takes a little longer to init for proper consistency

Reply:
Nothing against matching on name - it's just that some of the Bukkit enum names don't match the NMS names. Currently ordinal is the only way to match them. I could replace with a switch statement that falls back to names with special cases for the non-matching enum values?